### PR TITLE
Support py3.10 and remove py3.7

### LINKED ISSUE #368: [Bug]got error while running tutorial
# 🐛 Bug

<-- got error while running tutorial  -->

## To reproduce

1. run the tutorial.ipynb

** Stack trace/error message **
```
ERROR: Package 'pykale' requires a different Python: 3.10.11 not in '<3.10,>=3.7'
/content/pykale/examples/digits_dann
```

## Additional context
should update pykale current code or downgrade by using code 

#install python 3.9
!sudo apt-get update -y
!sudo apt-get install python3.8
!sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.8 1
!sudo update-alternatives --config python3
!apt-get install python3-pip
!python -m pip install --upgrade pip --user

then it will install successfully,
from config import get_cfg_defaults
but still get error: 
```
ModuleNotFoundError: No module named 'yacs' 
```

### PR DESCRIPTION
**Pull request by Mdnaimulislam**
Fixes #368.

### Description
Added new python version 3.10 and removed python version 3.7.

### Status
**Ready**

### Types of changes

- [ ] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] In-line docstrings updated.
- [ ] [Source for documentation at `docs`](https://github.com/pykale/pykale/tree/main/docs/source) manually updated for new API.

### CODE REVIEW

**Comment by XianyuanLiu** [comment]
It seems that Python 3.10 costs much more time in dependency installation than 3.8 and 3.9.

**Review by Mdnaimulislam — COMMENTED** [review]
It looks like the sklearn package (not supported by Python 3.10) is being used by one of our dependencies. See this https://github.com/scikit-learn/sklearn-pypi-package, where it says to track down which dependency is using sklearn. I have tried to find it for the last couple of days but haven't been able to track it down. So, for the time being, I have set the environment variable SKLEARN_ALLOW_DEPRECATED_SKLEARN_PACKAGE_INSTALL=True to avoid this error (as provided by the above URL I mentioned).

File: setup.py [hunk 1]
```diff
@@ -13,6 +14,9 @@
 
 # Key reference followed: https://github.com/pyg-team/pytorch_geometric/blob/master/setup.py
 
+# Set an environment variable to bypass the SKLEARN error message.
```
Comment (XianyuanLiu) [inline setup.py:17]: Please add the link to [sklearn-pypi-package](https://github.com/scikit-learn/sklearn-pypi-package#how-to-fix-the-error-for-the-main-use-cases) for further information, preventing those unfamiliar with this PR from getting confused about why this error was bypassed.

**Review by XianyuanLiu — CHANGES_REQUESTED** [review]
Thanks and well done! There are some minor enhancements. This bypassing may be tracked, and removed once it becomes unnecessary.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -19,10 +19,8 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest]
-        python-version: ["3.7", "3.8", "3.9"] # list of Python versions to test
+        python-version: ["3.8", "3.9", "3.10"] # list of Python versions to test
         exclude:
-          - os: windows-latest
-            python-version: "3.7"
           - os: windows-latest
             python-version: "3.8"
```
[the remark this answers is not in the document]
Comment (Mdnaimulislam) [reply .github/workflows/test.yml:25]: Okay, keeping only the Python 3.10 for simplify the test process.

**Comment by Mdnaimulislam** [comment]
I didn't have any idea that the error was being caused by the nilearn version. After my research, my assumption was that it was due to the setuptools version. I encountered a couple of places where the same problem arose because of setuptools. So yes, this problem is related to pip and setuptools. When I was trying to find the cause locally, I came across an error: "SetuptoolsDeprecation: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer." I am not entirely sure if this is connected to the problem we are having. I have changed the nilearn version; let's see if this solves our problem.

**Comment by XianyuanLiu** [comment]
There are still some Python 3.7 usages in [workflow and description](https://github.com/search?q=repo%3Apykale%2Fpykale%203.7&type=code). We'd better update them.

**Review by XianyuanLiu — APPROVED** [review]
LGTM

**Pull request merged**